UI: Hide Propspec bind if disabled

NickCloudAT opened the pull request:
This PR would "fix" https://github.com/TTT-2/TTT2/issues/1706

I'm just not sure: Is using globals here a good idea? I though about using a replicated cvar, but that would require at least a cl_propspec.lua file to be added.

Changelog will follow when finished.

TimGoll commented:
I see the problem here. I'd prefer not to use global bools. Maybe we can create a shared convar file where stuff like this could be moved to? hm.

NickCloudAT commented:
> I see the problem here. I'd prefer not to use global bools. Maybe we can create a shared convar file where stuff like this could be moved to? hm.

I just saw that here is a file called "sh_cvar_handler.lua".. Which would be exactly what we need, or am I wrong?
It's just not really used as of now.

TimGoll commented:
> > I see the problem here. I'd prefer not to use global bools. Maybe we can create a shared convar file where stuff like this could be moved to? hm.
> 
> I just saw that here is a file called "sh_cvar_handler.lua".. Which would be exactly what we need, or am I wrong? It's just not really used as of now.

Exactly. Not that you reminded me, I created this file with the purpose of moving ALL convars in there. But then ditched the idea because I thought it would be better to keep them in their modules.

But imho this file is fine for your use case

NickCloudAT commented:
> > > I see the problem here. I'd prefer not to use global bools. Maybe we can create a shared convar file where stuff like this could be moved to? hm.
> > 
> > 
> > I just saw that here is a file called "sh_cvar_handler.lua".. Which would be exactly what we need, or am I wrong? It's just not really used as of now.
> 
> Exactly. Not that you reminded me, I created this file with the purpose of moving ALL convars in there. But then ditched the idea because I thought it would be better to keep them in their modules.
> 
> But imho this file is fine for your use case

I changed it to what I thought about when mentioning the replicated cvar.
Only "problem" i see now is, that the keyhelp.lua seems to be loaded before the sh_cvar_handler file. So I cant cache the convar directly..

TimGoll commented on gamemodes/terrortown/gamemode/server/sv_propspec.lua:
```diff
@@ -9,9 +9,7 @@ local timer = timer
----
--- @realm server
-local cvPropspecToggle = CreateConVar("ttt_spec_prop_control", "1", { FCVAR_NOTIFY, FCVAR_ARCHIVE })
+local cvPropspecToggle = GetConVar("ttt_spec_prop_control")
```
just to be sure: did you test that the other file is loaded first?

NickCloudAT replied:
Yes, this works as expected.

It is loaded first:
cl main:
![image](https://github.com/user-attachments/assets/d8906876-7f73-4d84-b43d-732953c16bb4)

sv main:
![image](https://github.com/user-attachments/assets/b71439c0-e6c8-4bf2-addc-216c9d095314)

TimGoll commented:
> I changed it to what I thought about when mentioning the replicated cvar. Only "problem" i see now is, that the keyhelp.lua seems to be loaded before the sh_cvar_handler file. So I cant cache the convar directly..

That's fine, a convar lookup is not that expensive. I think it's good as it is now

NickCloudAT commented:
> > I changed it to what I thought about when mentioning the replicated cvar. Only "problem" i see now is, that the keyhelp.lua seems to be loaded before the sh_cvar_handler file. So I cant cache the convar directly..
> 
> That's fine, a convar lookup is not that expensive. I think it's good as it is now

The GMOD wiki on this is also not really helpful for me.. Either I understand something wrong, or the information on there is very convoluted.. https://wiki.facepunch.com/gmod/Global.GetConVar
It mentiones that it caches? But also that it does not?

TimGoll commented:
> > > I changed it to what I thought about when mentioning the replicated cvar. Only "problem" i see now is, that the keyhelp.lua seems to be loaded before the sh_cvar_handler file. So I cant cache the convar directly..
> > 
> > 
> > That's fine, a convar lookup is not that expensive. I think it's good as it is now
> 
> The GMOD wiki on this is also not really helpful for me.. Either I understand something wrong, or the information on there is very convoluted.. https://wiki.facepunch.com/gmod/Global.GetConVar It mentiones that it caches? But also that it does not?

Ah, interesting. It is cached, but it a bad way.

you can do it like this then:

```lua
-- top of the file
local cvPropspecToggle

-- in function
cvPropspecToggle = cvPropspecToggle  or GetConVar("ttt_spec_prop_control")
```

This is not perfect, but should be faster than the internal caching.

NickCloudAT commented:
> > > > I changed it to what I thought about when mentioning the replicated cvar. Only "problem" i see now is, that the keyhelp.lua seems to be loaded before the sh_cvar_handler file. So I cant cache the convar directly..
> > > 
> > > 
> > > That's fine, a convar lookup is not that expensive. I think it's good as it is now
> > 
> > 
> > The GMOD wiki on this is also not really helpful for me.. Either I understand something wrong, or the information on there is very convoluted.. https://wiki.facepunch.com/gmod/Global.GetConVar It mentiones that it caches? But also that it does not?
> 
> Ah, interesting. It is cached, but it a bad way.
> 
> you can do it like this then:
> 
> ```lua
> -- top of the file
> local cvPropspecToggle
> 
> -- in function
> cvPropspecToggle = cvPropspecToggle  or GetConVar("ttt_spec_prop_control")
> ```
> 
> This is not perfect, but should be faster than the internal caching.

Hm, yeah you are right. I added this, guess that's the best way in this case.

TimGoll commented:
awesome work, thank you!
